openedx/openedx-platform
# [BD-13] Let ModuleSystem use UserService

## Description

Initial step towards combining the ModuleSystem and DescriptorSystem classes.

Adds a shim which deprecates these ModuleSystem attributes, pulling them from the `user` service instead of requiring them in the constructor.

* `anonymous_student_id`
* `seed`
* `user_id`
* `user_is_staff`

This change is only a refactoring, and should not affect Learners, Course Authors, or anyone else using edx-platform.

## Supporting information

* [Content Core Platform Simplication project](https://openedx.atlassian.net/wiki/spaces/AC/pages/3091629503/Content+Core+Platform+Simplification)
* [BLENDED-113](https://openedx.atlassian.net/browse/BLENDED-113) (edX internal ticket)
* [FAL-2247](https://tasks.opencraft.com/browse/FAL-2247) (OpenCraft internal ticket)

Dependent on:
* https://github.com/edx/XBlock/pull/542

## Testing instructions

LMS: https://pr28440.sandbox.opencraft.hosting/
1. Login as a demo learner (`honor@example.com` / `edx`)
2. Browse around the Demo course and ensure that blocks render as expected, and that you can submit answers to problems.
3. Switch to the Legacy Course Experience and browse around in that too.

Studio: https://studio.pr28440.sandbox.opencraft.hosting/
1. Login as a demo staff (`staff@example.com` / `edx`)
2. Navigate to the Demo course and browse through the blocks. Ensure they render and edit as expected.
3. Check that you can submit answers to problems, and save changes to blocks.
4. Preview unpublished changes and ensure they render as expected.
4. Publish changes, and ensure they render in the LMS as expected.
 
## Deadline

None
